File: Non Linear Data Structures/Trees/C++/Tree Traversals/DFS/postorder.cpp
#include<iostream>
#include<stack>
using namespace std;

class Node {
    public :
        int data;
        Node* left;
        Node* right;

    Node(int data) {
        this->data = data;
        this->left = NULL;
        this->right = NULL;
    }
};

void postOrderTraversal(Node* root) {
    if(root == NULL) return;

    postOrderTraversal(root->left);
    postOrderTraversal(root->right);
    cout<<root->data<<"\t";
}

int main () {

    Node* root = new Node(10);
    root->left = new Node(11);
    root->right = new Node(21);

    postOrderTraversal(root);


    return 0;
}

File: Non Linear Data Structures/Trees/Python/Tree Traversals/DFS/postorder.py
class Node :
    def __init__(self , data):
        self.data = data
        self.left = None
        self.right = None
        
def postOrderTraversal(root) :
    if root == None :
        return
    postOrderTraversal(root.left)
    postOrderTraversal(root.right)
    print(root.data)

root = Node(10)
root.left = Node(11)
root.right = Node(22)

# print(root.data)
# print(root.left.data)
# print(root.right.data)

postOrderTraversal(root)